File: api/models/schemas/instrument.py
"""Pydantic v2 schemas for instruments."""

from __future__ import annotations

from datetime import datetime

from pydantic import BaseModel, ConfigDict


class InstrumentBase(BaseModel):
    symbol: str
    sec_type: str
    exchange: str
    currency: str = "USD"
    multiplier: float
    min_tick: float
    description: str | None = None


class InstrumentCreate(InstrumentBase):
    pass


class InstrumentUpdate(BaseModel):
    sec_type: str | None = None
    exchange: str | None = None
    currency: str | None = None
    multiplier: float | None = None
    min_tick: float | None = None
    description: str | None = None


class InstrumentResponse(InstrumentBase):
    id: int
    created_at: datetime | None = None
    updated_at: datetime | None = None

    model_config = ConfigDict(from_attributes=True)


class InstrumentsListResponse(BaseModel):
    total: int
    instruments: list[InstrumentResponse]

File: frontend/src/types/instrument.ts
export interface Instrument {
  id: number;
  symbol: string;
  sec_type: string;
  exchange: string;
  currency: string;
  multiplier: number;
  min_tick: number;
  description: string | null;
  data_from: string | null;
  data_to: string | null;
  created_at: string | null;
  updated_at: string | null;
}

export interface InstrumentsListResponse {
  total: number;
  instruments: Instrument[];
}

export interface ScanJobResponse {
  id: number;
  status: 'pending' | 'running' | 'completed' | 'failed';
  created_at: string;
  started_at: string | null;
  completed_at: string | null;
  error_message: string | null;
  results: { symbol: string; data_from: string | null; data_to: string | null }[] | null;
}
